File: interviews/leetcode/1431. Kid With Greatest Number of Candies/basic.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	// find the total number of candies
	maxCandies := maxVal(candies)

	// create the return array
	retArr := make([]bool, len(candies))

	// compute the new max value for each
	for i := 0; i < len(candies); i++ {
		retArr[i] = false
		if maxCandies <= candies[i]+extraCandies {
			retArr[i] = true
		}
	}
	return retArr
}

func maxVal(arr []int) int {
	mx := 0
	for i := 0; i < (len(arr)); i++ {
		mx = max(mx, arr[i])
	}
	return mx
}

File: interviews/leetcode/arrays/1431. Kid With Greatest Number of Candies/basic.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        # compute the largest number of candies
        max_candies = max(candies)

        # create the return array
        # [false] * len(candies)
        return [True if (candy_count+extraCandies>=max_candies) else False for candy_count in candies]
